### PR TITLE
feat: add manifests.yaml_stream impl at kclvm runtime.

### PR DESCRIPTION
**Pull request by Peefy**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

feat: #94 

* Add the runtime function to implement the customized output of YAML stream format manifests. 
* TODO: The corresponding implementation of the KCL UI layer includes function call and runtime impl integration and more UI test suites. @zong-zhe 
* TODO: More yaml encode options to impl. @Peefy 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

* unit test suites: kclvm/runtime/src/manifests/yaml.rs

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by ldxdl** [comment]
More UI test cases are needed with the new sys lib function.

**Comment by Peefy** [comment]
> More UI test cases are needed with the new sys lib function.

Right! The main change of this PR is the implementation of the runtime function. More language UI implementations and test cases need to be completed in another PR by @zong-zhe

**Comment by ldxdl** [comment]
More options to consider and support: 
+ Separator: which separator to use between YAML documents (defaults to '---')
+ Indent:  which kind of indentation to use when emitting (defaults to 2)
+ Width: The character width to use when folding text (defaults to 80)
+ UseFold: Force folding of text when emitting (defaults to false)
+ UseBlock: Force all text to be literal when emitting (defaults to false)
+ UseVersion: Display the YAML version when emitting (defaults to false)
+ UseHeader: Display the YAML header when emitting (defaults to false)
+ Encoding: Unicode format to encode with (defaults to :Utf8)

File: kclvm/runtime/src/manifests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#[allow(non_camel_case_types)]
+type kclvm_value_ref_t = ValueRef;
+
+/// The function is to serialize a list of KCL objects to YAML and output using the style with
+/// the `---` delimiter, and put it to the custom manifest output in the context.
+/// More information: https://github.com/KusionStack/KCLVM/issues/94
+///
+/// ```kcl, no run
+/// import manifests
+/// manifests.yaml_stream(values: [any], * , sort_keys: bool = False, ignore_private: bool = True, ignore_none: bool = False)
```
[the remark this answers is not in the document]
Comment (Peefy) [reply kclvm/runtime/src/manifests/mod.rs]: > Could we abstract all options to a 'opts'/'options' param for a better extension?

Yes. I have used a parameter `opts` with the `ManifestsYamlStreamOptions` schema type.

**Comment by Peefy** [comment]
> More options to consider and support:
> 
> * Separator: which separator to use between YAML documents (defaults to '---')
> * Indent:  which kind of indentation to use when emitting (defaults to 2)
> * Width: The character width to use when folding text (defaults to 80)
> * UseFold: Force folding of text when emitting (defaults to false)
> * UseBlock: Force all text to be literal when emitting (defaults to false)
> * UseVersion: Display the YAML version when emitting (defaults to false)
> * UseHeader: Display the YAML header when emitting (defaults to false)
> * Encoding: Unicode format to encode with (defaults to :Utf8)

This is a good suggestion. I have added some of the options, and some of them left todo because `serde_yaml` library does not support these capabilities, such as the indent option (Ref: https://github.com/dtolnay/serde-yaml/issues/337)

File: kclvm/runtime/src/manifests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            Err(err) => panic!(
+                "Invalid options in yaml_stream(), the error message is {}",
+                err
+            ),
+        },
+        None => YamlEncodeOptions::default(),
+    };
+
+    if let Some(value) = args.arg_i(0) {
+        self::yaml::encode_yaml_stream_to_manifests(ctx, &value, opts);
```
Comment (chai2010) [inline kclvm/runtime/src/manifests/mod.rs:68]: missing return?

**Review by ldxdl — APPROVED** [review]
lgtm

**Review by chai2010 — APPROVED** [review]
LGTM

**Pull request merged**